core/src/test/java/org/apache/iceberg/rest/requests/TestCreateNamespaceRequest.java

```diff
@@ -0,0 +1,105 @@
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.util.JsonUtil;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class TestCreateNamespaceRequest {
+
+  private static final ObjectMapper MAPPER = JsonUtil.withRESTComponents();
+
+  // Non-empty
+  private static final String JSON_REQUEST_STRING =
+      "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":{\"owner\":\"Hank\"}}";
+  private static final Namespace NAMESPACE = Namespace.of("accounting", "tax");
+  private static final Map<String, String> PROPERTIES = ImmutableMap.of("owner", "Hank");
+
+  // Empty - The response requires a non-empty namespace
+  private static final String JSON_REQUEST_NO_PROPERTIES =
+      "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":{}}";
+
+  @Test
+  public void testDeserialize() throws JsonProcessingException {
+    CreateNamespaceRequest expected = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE)
+        .withProperties(PROPERTIES)
+        .build();
+
+    CreateNamespaceRequest actual = MAPPER.readValue(JSON_REQUEST_STRING, CreateNamespaceRequest.class);
+
+    assertSame(expected, actual);
+  }
+
+  @Test
+  public void testDeserializeNoProperties() throws JsonProcessingException {
+    CreateNamespaceRequest expected = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE)
+        .build();
+
+    CreateNamespaceRequest actual = MAPPER.readValue(JSON_REQUEST_NO_PROPERTIES, CreateNamespaceRequest.class);
+
+    assertSame(expected, actual);
+  }
+
+  @Test
+  public void testSerialize() throws JsonProcessingException {
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE)
+        .withProperties(PROPERTIES)
+        .build();
+
+    Assertions.assertThat(MAPPER.writeValueAsString(req))
+        .isEqualTo(JSON_REQUEST_STRING);
+  }
+
+  @Test
+  public void testSerializeNoProperties() throws JsonProcessingException {
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE)
+        .build();
+
+    Assertions.assertThat(MAPPER.writeValueAsString(req))
+        .isEqualTo(JSON_REQUEST_NO_PROPERTIES);
+  }
+
+  private static void assertSame(CreateNamespaceRequest expected, CreateNamespaceRequest actual) {
+    Assertions.assertThat(actual)
+        .satisfies(req -> {
+          Assertions.assertThat(req.namespace()).isNotNull().isEqualTo(expected.namespace());
+          Assertions.assertThat(req.properties()).isNotNull().isEqualTo(expected.properties());
+        });
+
+    // Ensure that this function is updated if new fields are added to the class.
+    // Avoids adding equals / hashCode method just for test purposes.
+    String[] allFields = {"namespace", "properties"};
+    Assertions.assertThat(actual)
+        .hasOnlyFields(allFields);
+    Assertions.assertThat(expected)
+        .hasOnlyFields(allFields);
```

Comment:
I would put this assertion in a separate test. The result is going to be the same for all instances of the object, right? So we don't need to check both expected and actual, nor do we need to do it when comparing expected and actual.

Reply:
Good point. I'll do it in just a separate test, one time.